docs/taskcluster-yml-v1.md

```diff
@@ -0,0 +1,189 @@
+
+* `tasks_for` - defines the type of event, one of `github-push`,
+  `github-pull-request` or `github-release`.
+
+* `event` - the raw Github event; see
+  * [PushEvent](https://developer.github.com/v3/activity/events/types/#pushevent)
+  * [PullRequestEvent](https://developer.github.com/v3/activity/events/types/#pullrequestevent)
+  * [ReleaseEvent](https://developer.github.com/v3/activity/events/types/#releaseevent)
+
+* `now` - the current time, as a string; this is useful for reproducible `$fromNow` invocations (see JSON-e documentation)
+
+* `as_slugid` - a function which, given a label, will generate a slugid. Multiple calls with the same label for the same event will generate the same slugid, but different slugids in different events.  Use this to generate taskIds, etc.
+
+## Result
+
+After rendering, the resulting data structure should have a `tasks` property containing a list of task definitions. Each task definition should match the [task
+schema](https://docs.taskcluster.net/reference/platform/taskcluster-queue/docs/task-schema) as it will be passed nearly unchanged to `Queue.createTask`, The exception is that the provided task definition must contain a `taskId` field, which the service will remove and pass to `Queue.createTask` directly.
+
+# Policies
+
+The `policy` property defines policies for what is allowed on the repository. Policies are always read from the default branch (generally `master`) of the repository. This prevents a malicious contributor from changing the policy applied to a pull request in the pull request itself.
+
+Taskcluster-Github will always build pushes and releases.
+
+##### Pull Requests
+
+Most projects prefer to run tasks for each pull request, so that the review process can take into account the task results. But if your project requires some secret data or uses some expensive service to test a pull request, then you probably do not want to run tasks for pull requests written by arbitrary contributors, but would still like to run tasks for PRs by project collaborators.
+
+The `pullRequests` policy controls this behavior:
+
+* `public` -- tasks are created for all pull requests.
+
+* `collaborators` (the default) -- tasks are created if the user who made the pull request is a collaborator on the repository.
+Github [defines
+  collaborators](https://developer.github.com/v3/repos/collaborators/#list-collaborators) as 
+  "outside collaborators, organization members with access through team memberships, organization members with access through default organization permissions, and organization owners."
+
+Example:
+```yaml
+policy:
+  pullRequests: public
+```
+# Task definition
+ 
+ ##  Configuration in `extra.github`
+
+ ### Github Events
+
+You can modify a task definition so that it will only run for specific Github events, by setting `task.extra.github.events` to a list of event names:
```

Comment:
Instead of using `extra.github`, I think we should require users to do this themselves using JSON-e expressions.  `{$if: 'tasks_for == 'github-push' && event.head.branch == 'development', then: ..}`